### PR TITLE
Import the initial arcade bits

### PR DESCRIPTION
**Pull request by lambdageek**
- Pull the `eng/common` stuff from https://github.com/dotnet/arcade
- Set the project to build and version info
- Add `.dotnet` to `.gitignore`
- Use dotnet 5.0.102 to build

This is enough to get `eng/common/build.sh --restore --build --pack` to build the project.

The Version.props has all the right versions, but Versions.Details.xml is not correct.

### CODE REVIEW

**Comment by lambdageek** [comment]
@akoeplinger does this seem like a good start?

File: Directory.Build.props [hunk 1]
```diff
@@ -1,7 +1,49 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
+    <!--
+      When building using source-build the process is:
+      - Newtonsoft.Json versions 9.0.1 and 12.0.2 are built by source-build
+      - Version 12.0.2 is written to Version.props
+      - Arcade needs to use 9.0.1 so we need to override Version.props value here
+    -->
+    <NewtonsoftJsonVersion>9.0.1</NewtonsoftJsonVersion>
```
Comment (akoeplinger) [inline Directory.Build.props:28]: are you even using Newtonsoft.Json?

File: NuGet.config [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<configuration>
+  <solution>
+    <add key="disableSourceControlIntegration" value="true" />
+  </solution>
+  <packageSources>
+    <clear />
+    <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
+    <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
+    <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
+    <add key="vs-buildservices" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-buildservices/nuget/v3/index.json" />
```
Comment (akoeplinger) [inline NuGet.config:11]: you probably don't need this feed

File: eng/Build.props [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup>
+    <ProjectToBuild Include="$(MSBuildThisFileDirectory)..\hotreload-utils.proj" />
```
Comment (akoeplinger) [inline eng/Build.props:4]: if you directly add `src/hotreload-delta-gen/src/hotreload-delta-gen.csproj `here then you don't need `hotreload-utils.proj`

File: eng/Version.Details.xml [hunk 4]
```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Dependencies>
+  <ProductDependencies>
+    <Dependency Name="Microsoft.SymbolUploader.Build.Task" Version="1.1.156702">
```
Comment (akoeplinger) [inline eng/Version.Details.xml:4]: do you really need all of these dependencies?

File: eng/Versions.props [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    <!-- Base three-part version used for all outputs of the repo (assemblies, packages, vsixes) -->
+    <VersionPrefix>0.0.0</VersionPrefix>
+    <!-- Package pre-release label not including build number or the pre-release iteration-->
+    <PreReleaseVersionLabel>alpha</PreReleaseVersionLabel>
+    <!-- Package pre-release version iteration. Combines with the label to produce a final pre-release suffix. -->
+    <PreReleaseVersionIteration>0</PreReleaseVersionIteration>
+    <!-- Opt-out repo features -->
+    <UsingToolXliff>false</UsingToolXliff>
+    <!-- Libs -->
+    <CommandLineParserVersion>2.2.1</CommandLineParserVersion>
```
Comment (akoeplinger) [inline eng/Versions.props:14]: same here, it looks like this has a lot of unused/unnecessary dependencies

File: global.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+    "tools": {
+      "dotnet": "5.0.102"
+    },
+    "msbuild-sdks": {
+      "Microsoft.DotNet.Arcade.Sdk": "6.0.0-beta.21101.7",
+      "Microsoft.DotNet.Helix.Sdk": "6.0.0-beta.21101.7"
```
Comment (akoeplinger) [inline global.json:7]: remove this if you don't use helix

**Review by akoeplinger — COMMENTED** [review]
Looks good in general but can be cleaned up a bit :)

File: Directory.Build.props (hunk 1 above)
Comment (lambdageek) [reply Directory.Build.props:28]: No.  All the dependencies are wrong.  This is just copy/pasted from the dotnet/arcade repo.

Also I'm not actually using any of the `XYZVersion` properties in the .csproj.

I'm still figuring out DARC channels and subscriptions and where to get the packages I actually need.

File: eng/Build.props (hunk 3 above)
Comment (lambdageek) [reply eng/Build.props:4]: I like the traversal project.  I think there will eventually be other packages in here other than the delta generator.

**Comment by lambdageek** [comment]
Ok, using Versions props now. I'll merge this and then start on darc once I read the docs.

**Pull request merged**